graph/src/components/ethereum/adapter.rs

```diff
@@ -103,64 +146,124 @@ impl EthereumLogFilter {
         // First topic should be event sig
         match log.topics.first() {
             None => false,
-            Some(sig) => self
-                .contract_address_and_event_sig_pairs
-                .iter()
-                .any(|pair| match pair {
-                    // The `Log` matches the filter either if the filter contains
-                    // a (contract address, event signature) pair that matches the
-                    // `Log`...
-                    (Some(addr), s) => addr == &log.address && s == sig,
-
-                    // ...or if the filter contains a pair with no contract address
-                    // but an event signature that matches the event
-                    (None, s) => s == sig,
-                }),
+
+            Some(sig) => {
+                // The `Log` matches the filter either if the filter contains
+                // a (contract address, event signature) pair that matches the
+                // `Log`, or if the filter contains wildcard event that matches.
+                self.contracts_and_events_graph
+                    .all_edges()
+                    .any(|(s, t, ())| {
+                        let contract = LogFilterNode::Contract(log.address.clone());
```

Comment:
I'd create `contract` and `event` upfront to avoid creating it over and over again.

Reply:
Good catch.